### ISSUE TITLE
projects/core/howto/udp.rst should be properly formatted

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6943]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#6943|
|Type|task|
|Created|2014-02-03 14:57:49Z|
|Branch|https://github.com/twisted/twisted/tree/udp-docs-format-6943|

Lore2Sphinx outputs rST which is functionally correct, but isn't pretty in source. There's formatting errors which will need to be fixed up manually.

I wrote a [small script](https://gist.github.com/hawkowl/8771120) that tells me all of the branches with changes for the old Lore sources. As the procedure to port these is "run lore to sphinx on them", we can't touch them until those branches are merged without getting merge conflicts.

docs/projects/core/howto/udp.rst is not one of these conflicting files, so this ticket is about formatting it properly.

<details><summary>Searchable metadata</summary>

```
trac-id__6943 6943
type__task task
reporter__hawkowl hawkowl
priority__normal normal
milestone__Lore_to_Sphinx Lore to Sphinx
branch__branches_udp_docs_format_6943 branches/udp-docs-format-6943
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1391439469000000 1391439469000000
changetime__1391447857000000 1391447857000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [41526]) Branching to udp-docs-format-6943.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl removed owner|
|-|-|

Putting this up to review.

This is purely formatting changes - things like semantic newlines, spaces where they shouldn't be and rSTification where needed.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

Thanks!  Just a couple minor points:

  1. There's still quite a few lines with trailing whitespace.  Please clean those up as well.
  1. reST renders `--` as en dash.  In the sentence ''A connected UDP socket is slightly different from a standard one -- it can only send and receive datagrams to/from a single address, but this does not\
 in any way imply a connection.'' en dash is clearly incorrect.  So was hyphen, before, so this isn't exactly any worse.  Considering you're changing the punctuation anyway, though, consider changing it to a colon or a period.

Please merge after fixing those.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [41536]) Merge udp-docs-format-6943: projects/core/howto/udp.rst should be properly formatted

Author: hawkowl
Reviewer: exarkun
Fixes: [#6943](https://github.com/twisted/twisted/issues/6943)

Minor documentation changes, including reformatting to semantic newlines and minor grammar fixes.
